### PR TITLE
Update Read.me with clarifying language, SQL, "What is CockroachDB" 

### PR DESCRIPTION
**Pull request by jess-edwards**
Update Read.me per Simson's comments and some style changes:
- added "What is CockroachDB"  section
- removed 2014 Wired article
- updated Status
- find-replace Cockroach --> Cockroach DB
- find-replace Cockroach nodes --> RoachNodes
- added link to SQL branch
- in Architecture, changed "CockroachDB is a distributed database...." sentence to include SQL + link to SQL branch
- removed link to structured data api because the link is dead
- In "Design," updated the URL to tech talk slides to a more recent deck. I was getting an error message when opening the currently linked deck.

### CODE REVIEW

**Comment by mrtracy** [comment]
LGTM

File: README.md [hunk 1]
```diff
@@ -7,23 +7,26 @@
...
+- [Deploying CockroachDB in production](#deploying-cockroach-in-production)
 - [Getting in touch and contributing](#get-in-touch)
 - [Talks](#talks)
 - [Design](#design) and [Datastore Goal Articulation](#datastore-goal-articulation)
 - [Architecture](#architecture) and [Client Architecture](#client-architecture)
 
-[![WIRED on CockroachDB](/resource/doc/wired-preview.png?raw=true)](http://www.wired.com/2014/07/cockroachdb/)
+## What is CockroachDB
+
+CockroachDB is a distributed SQL database built on top of a key:value datastore. It was designed to support ACID transactional semantics and versioned values as first-class features. The primary design goal is global consistency and survivability, hence the name. CockroachDB aims to tolerate disk, machine, rack, and even datacenter failures with minimal latency disruption and no manual intervention. CockroachDB nodes (RoachNodes) are symmetric; a design goal is homogenous deployment (one binary) with minimal configuration.
```
Comment (vivekmenezes) [inline README.md]: CockroachDB is a distributed SQL database built on top of a transactional and consistent key:value store. The primary design goals are support for ACID transactions, horizontal scalability and survivability, hence the name. CockroachDB implements a Raft consensus algorithm for consistency. It aims to tolerate disk, machine, rack, and even datacenter failures with minimal latency disruption and no manual intervention. CockroachDB nodes (RoachNodes) are symmetric; a design goal is homogenous deployment (one binary) with minimal configuration.

File: README.md [hunk 2]
```diff
@@ -150,25 +153,22 @@ Once that's done, take a look at our [open issues](https://github.com/cockroachd
...
-For a quick design overview, see the [Cockroach tech talk slides](https://docs.google.com/presentation/d/1e3TOxImRg6_nyMZspXvzb2u43D6gnS5422vAIN7J1n8/edit?usp=sharing)
+For a quick design overview, see the [CockroachDB tech talk slides](https://docs.google.com/presentation/d/1tPPhnpJ3UwyYMe4MT8jhqCrE9ZNrUMqsvXAbd97DZ2E/edit#slide=id.p)
 or watch a [presentation](#talks).
 
 
-Cockroach is a distributed key/value datastore which supports ACID
-transactional semantics and versioned values as first-class
-features. The primary design goal is global consistency and
-survivability, hence the name. Cockroach aims to tolerate disk,
+CockroachDB is a distributed SQL database built on top of a key:value datastore. It was designed to support ACID transactional semantics and versioned values as first-class features. The primary design goal is global consistency and survivability, hence the name. CockroachDB aims to tolerate disk,
```
Comment (vivekmenezes) [inline README.md]: I think you can get rid of the "versioned values" part and combines these lines like how I've done above.

**Comment by petermattis** [comment]
This change looks like a good stop-gap to me, but I think Spencer should take a crack at rewriting the entire README to reflect the current reality, simplifying it and moving the design portions into the `docs` directory with appropriate links.

File: README.md [hunk 3]
```diff
@@ -280,26 +280,26 @@ sorted-order storage algorithm selection. Btree stores are
...
 ![Read vs. Write Optimization Spectrum](/resource/doc/read-vs-write.png?raw=true)
 
 ## Architecture
 
-Cockroach implements a layered architecture, with various
+CockroachDB implements a layered architecture, with various
 subdirectories implementing layers as appropriate. The highest level of
-abstraction is the SQL layer (currently not implemented). It depends
-directly on the [structured data API][5] ([structured/][6]). The structured
+abstraction is the [SQL layer][5], which is currently in development and depends
```
Comment (vivekmenezes) [inline README.md]: you need not mention that it is in development. The entire system is in development

File: README.md [hunk 4]
```diff
@@ -312,11 +312,11 @@ replicas.
 
 ## Client Architecture
 
-Cockroach nodes serve client traffic using a fully-featured key/value
+RoachNodes serve client traffic using a fully-featured key/value
```
Comment (vivekmenezes) [inline README.md]: using a fully-featured SQL API.

File: README.md [hunk 5]
```diff
@@ -150,25 +153,22 @@ Once that's done, take a look at our [open issues](https://github.com/cockroachd
...
+CockroachDB is a distributed SQL database built on top of a key:value datastore. It was designed to support ACID transactional semantics and versioned values as first-class features. The primary design goal is global consistency and survivability, hence the name. CockroachDB aims to tolerate disk,
 machine, rack, and even datacenter failures with minimal latency
-disruption and no manual intervention. Cockroach nodes are symmetric;
+disruption and no manual intervention. RoachNodes are symmetric;
 a design goal is one binary with minimal configuration and no required
 auxiliary services.
 
-Cockroach implements a single, monolithic sorted map from key to value
-where both keys and values are byte strings (not unicode). Cockroach
+CockroachDB implements a single, monolithic sorted map from key to value
```
Comment (vivekmenezes) [inline README.md]: Just above this paragraph we could add a paragraph about SQL. Perhaps you can write it in a different PR (change). No need to do it here.

**Comment by vivekmenezes** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (spencerkimball) [inline README.md]: :+1:

File: README.md [hunk 6]
```diff
@@ -34,7 +37,7 @@ If you don't want to use Docker,
 
 #### Bootstrap and talk to a single node
 
-Getting a Cockroach node up and running is easy. If you have the `cockroach` binary, skip over the next shell session. Most users however will want to run the following:
+Getting a RoachNode node up and running is easy. If you have the `cockroach` binary, skip over the next shell session. Most users however will want to run the following:
```
Comment (spencerkimball) [inline README.md]: s/RoachNode node/RoachNode/  (elide the duplicative "node")

File: README.md [hunk 7]
```diff
@@ -7,23 +7,26 @@
...
+
+CockroachDB is a distributed SQL database built on top of a key:value datastore. It was designed to support ACID transactional semantics and versioned values as first-class features. The primary design goal is global consistency and survivability, hence the name. CockroachDB aims to tolerate disk, machine, rack, and even datacenter failures with minimal latency disruption and no manual intervention. CockroachDB nodes (RoachNodes) are symmetric; a design goal is homogenous deployment (one binary) with minimal configuration.
 
 ## Status
 
-See our
+CockroachDB is currently in alpha. See our
 [Roadmap](https://github.com/cockroachdb/cockroach/issues/2132) and
-[Issues](https://github.com/cockroachdb/cockroach/issues)
+[Issues](https://github.com/cockroachdb/cockroach/issues) for a list of features planned or in development.
```
Comment (tamird) [inline README.md]: line wrapping in this doc is outta control

**Comment by spencerkimball** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (xiang90) [inline README.md]: > a Raft consensus algorithm

A multi-Raft consensus algorithm? (also you can link to the blog post?)

File: README.md (hunk 3 above)
Comment (jess-edwards) [inline README.md]: done

File: README.md (hunk 2 above)
Comment (jess-edwards) [inline README.md]: done

File: README.md (hunk 4 above)
Comment (jess-edwards) [inline README.md]: done

File: README.md (hunk 6 above)
Comment (jess-edwards) [inline README.md]: done

File: README.md (hunk 7 above)
Comment (jess-edwards) [inline README.md]: Will fix this in a different PR

File: README.md (hunk 1 above)
Comment (jess-edwards) [inline README.md]: @vivekmenezes fixed. 

@xiang90 Don't feel this it the right place for a link to the blog post, but will look for a better place to link to it.

File: README.md (hunk 5 above)
Comment (jess-edwards) [inline README.md]: Agreed - will address in a different PR.

**Pull request merged**